src/bin/juliainstaller.rs

```diff
@@ -388,6 +388,43 @@ pub fn main() -> Result<()> {
         return Ok(());
     }
 
+    if install_choices.modifypath {
+        // Now check whether we have the necessary permissions for all the files
+        // we want to modify.
+
+        let paths = find_shell_scripts_to_be_modified(false)?;
+
+        let mut failed_paths: Vec<PathBuf> = Vec::<PathBuf>::new();
+
+        for cur_path in paths {
+            let file_result = std::fs::OpenOptions::new()
+                .read(true)
+                .write(true)
+                .open(&cur_path);
+
+            if file_result.is_err() {
+                failed_paths.push(cur_path.clone());
+            }
+        }
```

Comment:
This will fail if the file doesn't exist at that location.

Reply:
If you look at how we construct the `paths` vector, we look for existing files and then only run this for loop for files that we found on disc. Admittedly, there is a race condition here that someone could delete the file between our check and this code here, but I think that is ok, it will just error and then they can try again.